src/functions-and-arrays.js

```diff
@@ -1,41 +1,142 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
-
-
+function maxOfTwoNumbers(num1, num2) {
+  if (num1 > num2) {
+    return num1;
+  } else {
+    return num2;
+  }
+}
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
-
+function findLongestWord(input) {
+  if (input.length === 0) {
+    return null;
+  }
+  let longestWord = '';
+  for (let i = 0; i < input.length; i++) {
+    if (longestWord.length < input[i].length) {
+      longestWord = input[i];
+    }
+  }
+  return longestWord;
+}
 
+console.log(findLongestWord(words));
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
-
+function sumNumbers(element) {
+  let sumElements = 0;
+  for (i = 0; i < element.length; i++) {
+    sumElements += element[i];
+  }
+  return sumElements;
+}
 
+console.log(sumNumbers(numbers));
 
 // Iteration #3.1 Bonus:
-function sum() {}
 
+const mixedArr = [6, 12, 'miami', 1, true, 'barca', '200', 'lisboa', 8, 10];
+
+function sum(mixedElements) {
+  let sumMixElements = 0;
+  for (i = 0; i < mixedElements.length; i++) {
+    if (typeof mixedElements[i] === 'number') {
+      sumMixElements += mixedElements[i];
+    } else if (typeof mixedElements[i] === 'string') {
+      sumMixElements += mixedElements[i].length;
+    } else if (typeof mixedElements[i] === 'boolean') {
+      if (mixedElements[i]) {
+        sumMixElements++;
+      } else {
+        sumMixElements += 0;
+      }
+    } else if (typeof mixedElements[i] === 'object' || typeof mixedElements[i] === 'array') {
+      throw new Error("Unsupported data type sir or ma'am");
+    }
+  }
+  return sumMixElements;
+}
 
+console.log(sum(mixedArr));
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
+function averageNumbers(element) {
+  if (element.length === 0) {
+    return null;
+  }
+  let sumElements = 0;
 
+  for (i = 0; i < element.length; i++) {
+    sumElements += element[i];
+  }
+
+  let averageElements = sumElements / element.length;
```

Comment:
You could have reused the `sum()` function written above (line 45) here! `averageElements = sum(element) / element.length` 

Also, since the value of `averageElements` won't be reassigned further in the code, best to declare it with a `const` instead of a `let`. 💯